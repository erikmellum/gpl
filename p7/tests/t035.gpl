rectangle rects[3];

on akey
{
    rects[0].x = 1;
    rects[0].y = 2;
    rects[0].w = 3;
    rects[0].h = 4;
}

on skey
{
    print("rects[0].x = " + rects[0].x);
    print("rects[0].y = " + rects[0].y);
    print("rects[0].w = " + rects[0].w);
    print("rects[0].h = " + rects[0].h);
}

