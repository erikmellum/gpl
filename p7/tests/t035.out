gpl.cpp::main()
  input file(tests/t035.gpl)
  random seed(42)
  read_keypresses_from_standard_input(true)
  dump_pixels(false)
  symbol_table(true)
  print_symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
game_object rects[0]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
game_object rects[1]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
game_object rects[2]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
gpl.cpp::main() Calling window->initialize().
gpl.cpp::main() Passing control to window->main_loop().
gpl[13]: rects[0].x = 1
gpl[14]: rects[0].y = 2
gpl[15]: rects[0].w = 3
gpl[16]: rects[0].h = 4

Printing the symbol table from signal_handler()
game_object rects[0]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(4),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(3),
    x = int(1),
    y = int(2)
  }
game_object rects[1]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
game_object rects[2]
  Rectangle
  {
    animation_block = animation_block(0),
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(10),
    proximity = int(4),
    red = double(1),
    rotation = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(0)
  }
